.github/workflows/docker-build-package.yml

```diff
@@ -0,0 +1,45 @@
+name: build docker image and push
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - master
+      - staging
+    tags:
+      - "*"
+
+env:
+  IMAGE_NAME: goxygen
+
+jobs:
+  push:
+    runs-on: ubuntu-latest
+    permissions:
+      packages: write
+      contents: read
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Build image
+        run: sudo docker build . --file Dockerfile --tag $IMAGE_NAME --label "runnumber=${GITHUB_RUN_ID}"
+
+      - name: Log in to registry
+        # This is where you will update the PAT to GITHUB_TOKEN
+        run: echo "${{ secrets.GITHUB_TOKEN }}" | docker login ghcr.io -u ${{ github.actor }} --password-stdin
+
+      - name: Push image
```

Comment:
I didn't test this part yet. Can you please tell me what tags would it produce and for what conditions? I want to make sure I understand the code correctly.